static/app/views/explore/charts/index.tsx

```diff
@@ -1,5 +1,134 @@
-interface ExploreChartsProps {}
+import styled from '@emotion/styled';
 
-export function ExploreCharts({}: ExploreChartsProps) {
-  return <div>TODO: visualize charts</div>;
+import {getInterval} from 'sentry/components/charts/utils';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import {DEFAULT_RELATIVE_PERIODS} from 'sentry/constants';
+import {CHART_PALETTE} from 'sentry/constants/chartPalette';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {tooltipFormatter} from 'sentry/utils/discover/charts';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import Chart, {ChartType} from 'sentry/views/insights/common/components/chart';
+import ChartPanel from 'sentry/views/insights/common/components/chartPanel';
+import {useSpanIndexedSeries} from 'sentry/views/insights/common/queries/useDiscoverSeries';
+import {CHART_HEIGHT} from 'sentry/views/insights/database/settings';
+import {useMetricsIntervalParam} from 'sentry/views/metrics/utils/useMetricsIntervalParam';
+import {Subtitle} from 'sentry/views/profiling/landing/styles';
+
+import {useChartType} from '../hooks/useChartType';
+import {useVisualize} from '../hooks/useVisualize';
+
+interface ExploreChartsProps {
+  query: string;
+}
+
+const exploreChartTypeOptions = [
+  {
+    value: ChartType.LINE,
+    label: t('Line'),
+  },
+  {
+    value: ChartType.AREA,
+    label: t('Area'),
+  },
+  {
+    value: ChartType.BAR,
+    label: t('Bar'),
+  },
+];
+
+// TODO: Update to support aggregate mode and multiple queries / visualizations
+export function ExploreCharts({query}: ExploreChartsProps) {
+  const [visualize] = useVisualize();
+  const [chartType, setChartType] = useChartType();
+  const {interval, setInterval, currentIntervalOptions} = useMetricsIntervalParam();
+
+  const pageFilters = usePageFilters();
+  const period = pageFilters.selection.datetime.period;
```

Comment:
This breaks when the user selects an absolute date range. If possible, try to use the `datetime` object directly.

Reply:
Removed it